libs/java/gcp_zts_creds/src/main/java/com/yahoo/athenz/creds/gcp/GCPSIACredentials.java

```diff
@@ -0,0 +1,270 @@
+                .setDefaultRequestConfig(config)
+                .build()) {
+
+            // Construct an HTTP POST request.
+            String postPayload = "{" +
+                    "\"domain\": " + OBJECT_MAPPER.writeValueAsString(athenzDomain) + "," +
+                    "\"service\": " + OBJECT_MAPPER.writeValueAsString(athenzService) + "," +
+                    "\"provider\": " + OBJECT_MAPPER.writeValueAsString(athenzProvider) + "," +
+                    "\"attestationData\": " + OBJECT_MAPPER.writeValueAsString(attestationData) + "," +
+                    "\"csr\": " + OBJECT_MAPPER.writeValueAsString(csr) +
+                    "}";
+            // LOG.debug("Getting identity from ZTS: requesting \"{}/instance\" with payload: {}", ztsUrl, postPayload);
+            HttpEntity httpEntity = new StringEntity(postPayload, ContentType.APPLICATION_JSON);
+            HttpUriRequest httpUriRequest = RequestBuilder.post()
+                    .setUri(ztsUrl + "/instance")
+                    .setEntity(httpEntity)
+                    .addHeader("Content-Type", "application/json")
+                    .build();
+
+            // Execute the request and process the response.
+            HttpEntity httpResponseEntity = null;
+            try (CloseableHttpResponse httpResponse = httpClient.execute(httpUriRequest)) {
+                int statusCode = httpResponse.getStatusLine().getStatusCode();
+                httpResponseEntity = httpResponse.getEntity();
+                if ((statusCode == 200) || (statusCode == 201)) {
+                    InstanceIdentity response = OBJECT_MAPPER.readValue(httpResponseEntity.getContent(), InstanceIdentity.class);
+                    // LOG.debug("Got identity from ZTS:    x509Certificate={}    x509CertificateSigner={}", OBJECT_MAPPER.writeValueAsString(response.x509Certificate), OBJECT_MAPPER.writeValueAsString(response.x509CertificateSigner));
+                    return response;
+                } else {
+                    final String errorBody = (httpResponseEntity == null) ? "<no response body>" : EntityUtils.toString(httpResponseEntity);
+                    throw new Exception("Unable to register instance with Athenz. HTTP status: " + statusCode + ". Response: " + errorBody);
+                }
+            } finally {
+                EntityUtils.consumeQuietly(httpResponseEntity);
+            }
+        }
+    }
+
+    /** Used to parse ZTS response */
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    private static class InstanceIdentity {
+        public String x509Certificate;
+        public String x509CertificateSigner;
+    }
+
+    static String ATTESTATION_DATA_URL_PREFIX = "http://metadata/computeMetadata/v1/instance/service-accounts/default/identity?format=full&audience=";
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final Logger LOG = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
```

Comment:
we have this LOG object defined yet all of the log statements are commented out. If we're not planning any log entries, let's remove this and the dependency otherwise, let's enable the log statements. Be careful about the identity token ones - let's make sure we don't have any tokens logged even in debug mode.

Reply:
...all logs reluctantly removed...   ;-)